datafusion/proto/src/physical_plan/mod.rs

```diff
@@ -692,7 +692,14 @@ impl AsExecutionPlan for PhysicalPlanNode {
                         }
                     })
                     .collect::<Result<Vec<_>, _>>()?;
-                Ok(Arc::new(SortPreservingMergeExec::new(exprs, input)))
+                let fetch = if sort.fetch == -1 {
```

Comment:
```suggestion
                let fetch = if sort.fetch < 0 {
```
Maybe this version is safer. Also `Sort` uses this paradigm